### DISCUSSION TITLE
Derive Selectable should derive Queryable by default

### DISCUSSION THREAD

**Discussion by Elrendio** [diesel-rs/diesel#3746] [Ideas]
Hey!

Recently I've been migrating Stockly's code base to use `#[derive(Selectable)]`. I'm not aware of use cases where a structure can implement `Selectable` without implementing `Queryable`. Therefore I think the `Selectable` should by default add a `Queryable` implementation with an optout as `#[diesel(no_impl_queryable)]`.

**Pros:**
- More ergonomics, less characters to type
- Less cryptiques errors when you've forgotten a Queryable implementation. The most problematique part about this is that the error doesn't happen when you add the Selectable derive but when you try to use the struct.

**Cons:**
- This is a breaking change and can therefore be released only in diesel 3 if there's a version 3. 
- This *hides* more the different concepts

**Comment by weiznich** [comment]
I think its reasonable to go that way, but I really don't want to do another major release soon. 

What we could do without breaking change is to introduce another derive that combines `Queryable` + Selectable` + deprecate the using derive  `Queryable` + `Selectable` on the same struct.
